src/Arcus.Security.Providers.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -101,12 +122,20 @@ public virtual async Task<Secret> GetSecretAsync(string secretName)
             Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name to request a secret in Azure Key Vault");
             Guard.For<FormatException>(() => !SecretNameRegex.IsMatch(secretName), "Requires a secret name in the correct format to request a secret in Azure Key Vault, see https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#objects-identifiers-and-versioning");
 
+            IKeyVaultClient keyVaultClient = await GetClientAsync();
+
             try
             {
-                IKeyVaultClient keyVaultClient = await GetClientAsync();
                 SecretBundle secretBundle =
                     await ThrottleTooManyRequestsAsync(
-                        () => keyVaultClient.GetSecretAsync(VaultUri, secretName));
+                        async () =>
+                        {
+                            Logger.LogTrace("Getting a secret from Azure Key Vault...");
+                            SecretBundle bundle = await keyVaultClient.GetSecretAsync(VaultUri, secretName);
```

Comment:
Add dependency tracking here as well please, but that should be an opt-in/opt-out.

(follow-up PR is ok)